integration.js

```diff
@@ -119,6 +159,51 @@ async function downloadThumbnail(authToken, thumbnailUrl) {
   return `data:image/png;charset=utf-8;base64,${Buffer.from(response.data, 'binary').toString('base64')}`;
 }
 
+const MIME_EXPORT_TYPES = {
+  'application/vnd.google-apps.presentation': 'application/vnd.oasis.opendocument.presentation',
+  'application/vnd.google-apps.spreadsheet': 'text/csv',
+  'application/vnd.google-apps.document': 'application/vnd.oasis.opendocument.text',
+  'application/pdf': 'application/vnd.oasis.opendocument.presentation',
+  'application/vnd.google-apps.script': 'application/vnd.google-apps.script+json'
+};
+
+async function getFileContent(drive, file) {
+  const mimeType = MIME_EXPORT_TYPES[file.mimeType] || file.mimeType;
+  if (mimeType.includes('image') || mimeType.includes('jam') || mimeType.includes('jam')) return;
+
+  let requestResultBuffer;
+  try {
+    requestResultBuffer = await drive.files.export(
+      {
+        fileId: file.id,
+        mimeType: mimeType
+      },
+      {
+        responseType: 'arraybuffer'
+      }
+    );
+  } catch (error) {}
+  try {
+    requestResultBuffer = await drive.files.get({ fileId: file.id, alt: 'media' }, { responseType: 'arraybuffer' });
```

Comment:
@penwoodjon  I see that `requestResultBuffer` is assigned right before this but then not used.  Is the previous call necessary?  It seems like the call to `await drive.files.export` is just getting overwritten here.

Reply:
So this works as if only one of these requests (files.get/files.exports) will work without erroring out.  I don't remember all of the details, but I do remember realizing the requests to figure out which request function to use was more intensive than just running both requests and letting whichever one success win out and pass along the content.  If both fail, that means the service account doesn't have permissions and no content is returned.